packages/shared/lib/core/wallet/utils/convertToRawAmount.ts

```diff
@@ -19,13 +19,15 @@ export function convertToRawAmount(amount: string, tokenMetadata: TokenMetadata,
 function convertToRawAmountFromMetadata(amount: number, tokenMetadata: TokenMetadata, selectedUnit: string): Big {
     if (tokenMetadata?.standard === TokenStandard.BaseToken) {
         if (tokenMetadata.useMetricPrefix) {
-            return Big(amount * IOTA_UNIT_MAP?.[selectedUnit?.substring(0, 1)] ?? 0)
+            return Big(amount).mul(Big(10).pow(IOTA_UNIT_MAP?.[selectedUnit?.substring(0, 1)]?.decimalPlaces ?? 0))
         } else {
             if (selectedUnit === tokenMetadata.unit) {
                 const decimals = Math.min(tokenMetadata.decimals, MAX_SUPPORTED_DECIMALS)
                 return Big(amount).mul(Big(10).pow(decimals))
             } else if (selectedUnit === tokenMetadata.subunit) {
                 return Big(amount)
+            } else {
+                return undefined
```

Comment:
Perhaps `-1` might be better to return here since it's still a number but not a sensible amount that we'd have input to this function.

Reply:
Agreed. Also changed the other logic arm that returns undefined to match your suggestion. There are a few times in the code where we chain calls, like convertToRawAmount(stuff, stuff).additionalFunctionCallHere(), that would have caused an error if we returned undefined.